compiler/passes/src/type_checking/mod.rs

```diff
@@ -40,8 +40,62 @@ impl<'a, N: Network> Pass for TypeChecker<'a, N> {
     type Output = Result<(SymbolTable, StructGraph, CallGraph)>;
 
     fn do_pass((ast, handler, st, tt, max_depth, await_checking): Self::Input) -> Self::Output {
+        // get program name without consuming AST
+        let program_name = &ast
```

Comment:
Nit. This logic should be moved out of the `TypeChecker` pass and into here: https://github.com/ProvableHQ/leo/blob/mainnet/compiler/compiler/src/compiler.rs#L157
This is to ensure that output logic is contained in one place.

Reply:
Moved the logic to compiler as requested, it's better now, thanks about that.